### ISSUE TITLE
two types of ErrorCode ?

### ISSUE THREAD

**Issue by bogdan-sava** [odpi/egeria#2439]
I have noted that some of the  ErrorCode enums are having the fields:

```
    private int    httpErrorCode;
    private String errorMessageId;
    private String errorMessage;
    private String systemAction;
    private String userAction;
```

and others have 

```
    private String                     logMessageId;
    private OMRSAuditLogRecordSeverity severity;
    private String                     logMessage;
    private String                     systemAction;
    private String                     userAction;
```

Do we need two types of error codes?
Should the OMRSAuditLogRecordSeverity be field in all the audit log codes? Or none, and be passed on the log method?

I am asking in order to create a interface for ErrorCodes with default implementation for getFormattedMessage and simplify the usage of audit code and audit logs without change the current functionality.

**Comment by cmgrote** [comment]
These are purposely different, I believe.  The first are error codes (strictly for errors), and the latter are for audit records (which could be errors, but also informational, etc).

**Comment by bogdan-sava** [comment]
They can be, but still making no sense to have the OMRSAuditLogRecordSeverity field in the enum but the methods that are doing the logRecord and logException are having it as param.
In other words all the codes can be used with a different severity than the one in the field.

In this case we need to enforce the methods to log with the severity from the code, which can be done by refactoring the methods to get as param only the interface, or remove the severity from the codes as making no sense to have it there with only the purpose of using it as param.

If we want to keep two types of codes, we can have 2 interfaces and different log methods.  For the ones without severity, they will have an param to specify the severity, and for the one with severity field only will use it .

**Comment by cmgrote** [comment]
> They can be, but still making no sense to have the OMRSAuditLogRecordSeverity field in the enum but the methods that are doing the logRecord and logException are having it as param.
In other words all the codes can be used with a different severity than the one in the field.

Agreed -- I'd keep the severity in the code / enum itself, and not pass it as a parameter (makes using it simpler and less repetitive).

Also linking this to #2302 for the discussion there (as I assume these are the same topic?)...

**Comment by mandy-chessell** [comment]
The 2 types of enums are for different objects.

One is for exceptions - these are the error codes and include the HTTP response status.
The other is for the audit log messages - these are the audit codes and they have the OMRSAuditLogRecordSeverity in it.

If there are different styles for either or these two cases they need to be changed.

It is important that the OMRSAuditLogSeverity is part of the message definition and not left to the code that uses it because it defines the class of situation and to some extent who is likely to read the message description.

There are only 2 types of logging interfaces for the audit log - one to pass an exception and one that does not pass the exception.

The number of OMRSAuditLogRecordSeverity value is a constantly growing list as we look at more operational requirements.  There is more to the audit log than just is it logging an exception or is it informational.  There are some that require special routing (such as the security ones) and some that require operator action.  Having the messages and their severity externalized means that orgs can use the externalize message sets in their preparation/training/ops manuals to be ready to respond to different operational situations.

As I said in the other issue.  This area is being refactored as part of the GAF and I will incorporate some of these ideas in the refactorying.

I agree with @cmgrote that this is all part of the #2302 issue.

**Comment by bogdan-sava** [comment]
Closed in favor of  #2302 issue.
